apps/trading/client-pages/portfolio/account-history-container.tsx

```diff
@@ -0,0 +1,245 @@
+import { useAssetsDataProvider } from '@vegaprotocol/assets';
+import {
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuTrigger,
+  Splash,
+  Toggle,
+} from '@vegaprotocol/ui-toolkit';
+import { AccountTypeMapping } from '@vegaprotocol/types';
+import { PriceChart } from 'pennant';
+import 'pennant/dist/style.css';
+
+enum DateRange {
+  RANGE_1D = '1D',
+  RANGE_7D = '7D',
+  RANGE_1M = '1M',
+  RANGE_3M = '3M',
+  RANGE_1Y = '1Y',
+  RANGE_YTD = 'YTD',
+  RANGE_ALL = 'All',
+}
+
+const calculateStartDate = (range: string): string | undefined => {
+  const now = new Date();
+  switch (range) {
+    case DateRange.RANGE_1D:
+      return new Date(now.setDate(now.getDate() - 1)).toISOString();
+    case DateRange.RANGE_7D:
+      return new Date(now.setDate(now.getDate() - 7)).toISOString();
+    case DateRange.RANGE_1M:
+      return new Date(now.setMonth(now.getMonth() - 1)).toISOString();
+    case DateRange.RANGE_3M:
+      return new Date(now.setMonth(now.getMonth() - 3)).toISOString();
+    case DateRange.RANGE_1Y:
+      return new Date(now.setFullYear(now.getFullYear() - 1)).toISOString();
+    case DateRange.RANGE_YTD:
+      return new Date(now.setMonth(0)).toISOString();
+    default:
+      return undefined;
+  }
+};
+
+const dateRangeToggleItems = [
+  DateRange.RANGE_1D,
+  DateRange.RANGE_7D,
+  DateRange.RANGE_1M,
+  DateRange.RANGE_3M,
+  DateRange.RANGE_1Y,
+  DateRange.RANGE_YTD,
+  DateRange.RANGE_ALL,
+].map((range) => ({ label: t(range), value: range }));
+
+export const AccountHistoryContainer = () => {
+  const { pubKey } = useVegaWallet();
+  const { data: assets } = useAssetsDataProvider();
+
+  if (!pubKey) {
+    return <Splash>Connect wallet</Splash>;
+  }
+
+  if (!assets) {
+    return <Splash>Loading...</Splash>;
+  }
+
+  return <AccountHistoryManager pubKey={pubKey} assetData={assets} />;
+};
+
+const AccountHistoryManager = ({
+  pubKey,
+  assetData,
+}: {
+  pubKey: string;
+  assetData: AssetFieldsFragment[];
+}) => {
+  const [range, setRange] = useState<string>('All');
+  const [asset, setAsset] = useState<AssetFieldsFragment>();
+
+  const [accountType, setAccountType] = useState<Schema.AccountType>(
+    Schema.AccountType.ACCOUNT_TYPE_GENERAL
+  );
+
+  const variablesForOneTimeQuery = useMemo(
+    () => ({
+      partyId: pubKey,
+    }),
+    [pubKey]
+  );
+
+  const assetsWithBalanceHistory = useAccountsWithBalanceQuery({
+    variables: variablesForOneTimeQuery,
+    skip: !pubKey,
+  });
+
+  const assetsWithBalance = useMemo(
+    () =>
+      assetsWithBalanceHistory.data?.balanceChanges.edges.map(
+        (e) => e?.node.assetId
+      ) || [],
+    [assetsWithBalanceHistory.data?.balanceChanges.edges]
+  );
+
+  const assets = useMemo(() => {
+    const assetsFiltered = assetData
+      .filter((a) => assetsWithBalance.includes(a.id))
+      .sort((a, b) => a.name.localeCompare(b.name));
+    if (assetsFiltered && !asset) {
+      setAsset(assetsFiltered[0]);
```

Comment:
chore (blocking): its pretty non standard react practice to setState from within a useMemo. I'd suggest moving the useState below the memo and passing in a default value from the result of the useMemo

Reply:
updated in the last commit. still if I do that it won't select the asset by default. we can leave it with `Select asset` and then the user can choose